mailbox/opensearch/src/main/java/org/apache/james/mailbox/opensearch/OpenSearchMailboxConfiguration.java

```diff
@@ -36,46 +36,55 @@ public static class Builder {
         private Optional<IndexAttachments> indexAttachment;
         private Optional<IndexHeaders> indexHeaders;
 
+        private Optional<Boolean> messageIndexNewBehavior;
+
         Builder() {
             indexMailboxName = Optional.empty();
             readAliasMailboxName = Optional.empty();
             writeAliasMailboxName = Optional.empty();
             indexAttachment = Optional.empty();
             indexHeaders = Optional.empty();
+            messageIndexNewBehavior = Optional.empty();
         }
 
-        Builder indexMailboxName(Optional<IndexName> indexMailboxName) {
+        public Builder indexMailboxName(Optional<IndexName> indexMailboxName) {
             this.indexMailboxName = indexMailboxName;
             return this;
         }
 
-        Builder readAliasMailboxName(Optional<ReadAliasName> readAliasMailboxName) {
+        public Builder readAliasMailboxName(Optional<ReadAliasName> readAliasMailboxName) {
             this.readAliasMailboxName = readAliasMailboxName;
             return this;
         }
 
-        Builder writeAliasMailboxName(Optional<WriteAliasName> writeAliasMailboxName) {
+        public Builder writeAliasMailboxName(Optional<WriteAliasName> writeAliasMailboxName) {
             this.writeAliasMailboxName = writeAliasMailboxName;
             return this;
         }
 
-        Builder indexAttachment(IndexAttachments indexAttachment) {
+        public Builder indexAttachment(IndexAttachments indexAttachment) {
             this.indexAttachment = Optional.of(indexAttachment);
             return this;
         }
 
-        Builder indexHeaders(IndexHeaders indexHeaders) {
+        public Builder indexHeaders(IndexHeaders indexHeaders) {
             this.indexHeaders = Optional.of(indexHeaders);
             return this;
         }
 
+        public Builder messageIndexNewBehavior(Boolean messageIndexNewBehavior) {
+            this.messageIndexNewBehavior = Optional.ofNullable(messageIndexNewBehavior);
+            return this;
+        }
+
         public OpenSearchMailboxConfiguration build() {
             return new OpenSearchMailboxConfiguration(
                 indexMailboxName.orElse(MailboxOpenSearchConstants.DEFAULT_MAILBOX_INDEX),
                 readAliasMailboxName.orElse(MailboxOpenSearchConstants.DEFAULT_MAILBOX_READ_ALIAS),
                 writeAliasMailboxName.orElse(MailboxOpenSearchConstants.DEFAULT_MAILBOX_WRITE_ALIAS),
                 indexAttachment.orElse(IndexAttachments.YES),
-                indexHeaders.orElse(IndexHeaders.YES));
+                indexHeaders.orElse(IndexHeaders.YES),
+                messageIndexNewBehavior.orElse(DEFAULT_MESSAGE_INDEX_NEW_BEHAVIOR));
```

Comment:
Please default to the previous behaviour unless you are SURE I mean absolutly SURE that it is what is needed!

Reply:
Yes, it is, 
DEFAULT_MESSAGE_INDEX_NEW_BEHAVIOR = false